linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -1087,7 +1082,11 @@ object ClassDefChecker {
       copy(hasNewTarget = hasNewTarget)
 
     def withThisType(thisType: Type): Env =
-      copy(thisType = thisType)
+      withLocal(LocalDef(LocalName.This, thisType, mutable = false))
+
+    def withMaybeThisType(static: Boolean, thisType: Type): Env =
```

Comment:
Nit: `static` should maybe better be called `hasThis` here? (or `omitThis`).

Reply:
Oops, yes. Fixed. (and rebased just for safety)